Ark.Tools.ResourceWatcher/ResourceWatcher.cs

```diff
@@ -415,6 +418,89 @@ public class ProcessContext
         public ResultType? ResultType { get; set; }
         public int? Index { get; set; }
         public int? Total { get; set; }
+
+        public bool IsNewResource(out Tuple<string,LocalDateTime?,LocalDateTime?> changed)

```

Comment:
use a named Tuple for readability.

(LocalDateTime last, LocalDateTime current, string source)? nullable